Formula/jupyter.rb

```diff
@@ -0,0 +1,296 @@
+    sha256 "db0904a28253cfe53e7dedc765c71596f3c53bb8a866ae50123320ec1a7b73fd"
+  end
+
+  resource "traitlets" do
+    url "https://files.pythonhosted.org/packages/a5/98/7f5ef2fe9e9e071813aaf9cb91d1a732e0a68b6c44a32b38cb8e14c3f069/traitlets-4.3.2.tar.gz"
+    sha256 "9c4bd2d267b7153df9152698efb1050a5d84982d3384a37b2c1f7723ba3e7835"
+  end
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/55/11/e4a2bb08bb450fdbd42cc709dd40de4ed2c472cf0ccb9e64af22279c5495/wcwidth-0.1.7.tar.gz"
+    sha256 "3df37372226d6e63e1b1e1eda15c594bca98a22d33a23832a90998faa96bc65e"
+  end
+
+  resource "webencodings" do
+    url "https://files.pythonhosted.org/packages/0b/02/ae6ceac1baeda530866a85075641cec12989bd8d31af6d5ab4a3e8c92f47/webencodings-0.5.1.tar.gz"
+    sha256 "b36a1c245f2d304965eb4e0a82848379241dc04b865afcc4aab16748587e1923"
+  end
+
+  resource "widgetsnbextension" do
+    url "https://files.pythonhosted.org/packages/0a/09/083dbdbd6ba52004d6ee880877a7b3f6143f389743d20057e1eda3af3582/widgetsnbextension-2.0.0.tar.gz"
+    sha256 "566582a84642d0c0f78b756a954450a38a8743eeb8dad04b7cab3ca66f455e6f"
+  end
+
+  def install
+    ENV["JUPYTER_PATH"] = etc/"jupyter"
+    xy = Language::Python.major_minor_version "python"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python#{xy}/site-packages"
+
+    # gather packages to link based on options
+    linked = %w[jupyter_core jupyter_client nbformat ipykernel jupyter_console
+                nbconvert notebook qtconsole]
+    dependencies = resources.map(&:name).to_set - linked
+
+    if build.with?("notebook") && build.without?("nbconvert")
+      dependencies << "nbconvert"
+    end
+
+    linked.delete "jupyter_console" if build.without? "console"
+    linked.delete "nbconvert" if build.without? "nbconvert"
+    linked.delete "notebook" if build.without? "notebook"
+    linked.delete "qtconsole" if build.without? "qtconsole"
+
+    # install dependent packages
+    dependencies.each do |r|
+      resource(r).stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    # install main packages and link with env script
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
+    linked.each do |r|
+      resource(r).stage do
+        system "python", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :JUPYTER_PATH => ENV["JUPYTER_PATH"], :PYTHONPATH => ENV["PYTHONPATH"])
+
+    # remove bundled kernel
+    rm_rf Dir["#{libexec}/share/jupyter/kernels"]
+  end
+
+  def caveats; <<-EOS.undent
+    Additional kernels can be installed into the shared jupyter directory
+      #{etc/"jupyter"}
+    EOS
+  end
+
+  test do
+    assert_match "python2", shell_output("#{bin}/jupyter kernelspec list")
```

Comment:
Yeah it should definitely possible! I originally didn't to ensure the test wasn't dependent on any `option`s, but I'll make it more thorough for the default case 🙂